### PR TITLE
Fixes some beefman-related bugs and reworks the blood mechanics of beefmen

### PR DESCRIPTION
**Pull request by ATH1909**
## About The Pull Request

Dismembering a beefman's head no longer deletes their brain by turning their head into a slab of meat.

The three traumas that all beefman start each round with now have a resilience of TRAUMA_RESILIENCE_ABSOLUTE, so they can't just be removed by neurine or brain surgery anymore. When you take the red pill, you stay woke for life.

Salt is now a bit more scary for beefmen. Salt and saltpetre now apply their dehydration effects to beefmen immediately upon application, instead of gradually over time (there is, fortunately, now a cap of 100 for the value of dehydrate). Nonzero values of the dehydrate variable now cause more bleeding than they did before. Walking barefoot across a pile of salt also dehydrates you if you're a beefman. All of these salt interactions will also blur your vision and force you to scream (due to the pain).

The special bleeding mechanics of beefmen have been reworked. While the amount of burn damage (to organic body parts) that a beefman has still factors into how much they bleed, the temperature that a beefman's body is at no longer does. Also, beefmen are properly affected by things like heparin (and other sources of bleed_rate increases) now, as their special bleeding mechanics can no longer set their bleed_rate to an amount lower than what it already is.

The rate at which beefmen regenerate blood has also been slightly reduced, but in compensation, **the special beefman extra blood regeneration now works while the beefman is bleeding**. This means that beefmen no longer have to welder themselves or chug iron pills at shiftstart in order to not bleed out 10 minutes into the round, and in fact, they can maintain a stable blood level indefinitely while at 0 burn damage.

## Why It's Good For The Game

The head fix means that a beefman will no longer be almost permanently removed from the round (outside of podcloning and the like) if their head gets cut off (by, say, a medical doctor trying to perform a headswap maneuver).

Neurine shouldn't be taking away racial features. Also, those three traumas won't count towards the cap for TRAUMA_RESILIENCE_BASIC traumas (per brain) anymore, which frees up space in their brains for more W A C K Y special (and basic) traumas.

The changes to salt-related mechanics were made because I felt like splashing salt on a beefman should do SOMETHING, seeing as how they're weak to salt and all.

The changes to the blood mechanics were made for multiple reasons:

1. New players who choose the beefman race usually end up dead 10 minutes into the round, without much idea of why they bled out or how to avoid bleeding out the next time they play as a beefman. Yes, there's a guide to the racial features of beefmen, but how is a new player supposed to know where to find that guide, or even know that that guide exists (and is nigh-necessary to surviving as a beefman) in the first place?

2. The current beefman blood mechanics encourage you to NOT embrace your squicky, bleed-y nature, and to instead just welder yourself at roundstart and not bleed anywhere at all. Now you can bleed everywhere with pride, confident that you won't need a constant flow of iron pills to keep you from bleeding out. Or, if you want to be a beefman for the brain traumas, extra speed, etc., you can instead welder yourself to halt your racial bleeding (until your burn damage gets healed), just like you could before this PR.

tldr;
You won't see new beefman players crawl into medbay with half of their blood missing anymore, and janitors will have a lot more work to do now.

## Changelog
:cl: ATHATH
balance: Dismembering a beefman's head no longer deletes their brain by turning their head into a slab of meat.
balance: Beefmen can no longer remove their 3 roundstart brain traumas with neurine or brain surgery (or anything else short of admin intervention or a race change).
balance: Some changes were made to the way that beefmen bleed. The tldr; of it is that beefmen can now regenerate their blood while they're bleeding, so they no longer need to chug iron pills or welder themselves at roundstart in order to stay alive.
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
Note that this PR hasn't been tested yet; I've been having a busy week ~~and have been coding to procrastinate doing my college homework~~.

**Comment by remove32** [comment]
@TheSwain looking for your input on this one

**Comment by TheSwain** [comment]
I've been a bad boy. I'm gonna review this now.

**Comment by TheSwain** [comment]
Okay @ATH1909 , gonna un-stale this so I can look at it a bit more. I'm totally with you on the head removal, and I'm gonna get those trauma "resilience" flags in too.

Is there any reason you moved the salt chemistry code to the saltpetre's .dm instead of keeping it in the beefmen species .dm? The reason it was in there was, as always, to mess with TG's files & code as little as possible.

**Comment by ATH1909** [comment]
> Okay @ATH1909 , gonna un-stale this so I can look at it a bit more. I'm totally with you on the head removal, and I'm gonna get those trauma "resilience" flags in too.
> 
> Is there any reason you moved the salt chemistry code to the saltpetre's .dm instead of keeping it in the beefmen species .dm? The reason it was in there was, as always, to mess with TG's files & code as little as possible.

I moved it because I moved the code for salt+beefman reactions out as well, since handle_chemicals can't do things like make a race react to being splashed (or the like) by a reagent. Salt already has a reaction_mob() reaction (that I had to modify to incorporate this beefman interaction), and if /tg/ adds a reaction_mob() effect for saltpetre or something, I figure that it'd be better to just get a merge conflict than it'd be to have two definitions of a reaction_mob() proc for saltpetre.

**Comment by remove32** [comment]
we're still waiting on swain to review this i think

**Comment by ATH1909** [comment]
I think I'm just gonna close this; the wounds PR is gonna screw with bleeding and such and require a beefman rework anyway, I think.
